libr/debug/p/native/linux/linux_coredump.c

```diff
@@ -463,160 +433,148 @@ static void clean_maps(linux_map_entry_t *h) {
-		/* Check if the map comes from the kernel (vsyscall, vvar, vdso) (they are always dumped, but not vvar) */
+	ret = r_debug_map_sync (dbg);
+	if (!ret) return NULL;
+	r_list_foreach (dbg->maps, iter, map) {
+		linux_map_entry_t *pmentry = R_NEW0 (linux_map_entry_t);
+		if (!pmentry) goto error;
+		pmentry->start_addr = map->addr;
+		pmentry->end_addr = map->addr_end;
+		pmentry->name = strncmp (map->name, "unk", strlen ("unk")) ? strdup (map->name) : NULL;
+		/* Since RDebugMap does not provide an offset field (offset within the file), and the perms field
+			does not contain if is share or private, we need to check that ourself */
+		get_map_perms_and_offset (pmentry->start_addr, pmentry->end_addr, buff_maps, &flags_perm, &offset);
+		pmentry->perms = flags_perm;
+		pmentry->offset = offset;
+		pmentry->kernel_mapping = false;
+		
+		/* Check if is a kernel mapping */
 		if (pmentry->name && is_a_kernel_mapping (pmentry->name)) {
 			pmentry->anonymous = pmentry->kernel_mapping = true;
-			eprintf ("kernel_mapping: %d\n", pmentry->anonymous);
 		} else {
 			/* Check if map has anonymous content by checking Anonymous and AnonHugePages */
-			is_anonymous = has_map_anonymous_content (buff_smaps, start_addr, end_addr);
-			eprintf ("has_map_anonymous_content: %d\n", is_anonymous);
-			/* Check if pathname has a (deleted) part. Actually what kernel does is: file_inode(vma->vm_file)->i_nlink == 0 */
+			is_anonymous = has_map_anonymous_content (buff_smaps, pmentry->start_addr, pmentry->end_addr);
 			if (!is_anonymous && pmentry->name) {
-				is_anonymous = has_map_deleted_part (pmentry->name);
-				eprintf ("has_map_deleted_part called: %d\n", is_anonymous);
+				is_anonymous = is_deleted = has_map_deleted_part (pmentry->name);
 			}
 			pmentry->anonymous = is_anonymous;
 		}
-
-		if (!pmentry->kernel_mapping) {
-			if (pmentry->name && strcmp (pmentry->name, "[stack]") && strcmp (pmentry->name, "[heap]")) {
-				if (!pmentry->kernel_mapping)
-					pmentry->file_backed = true;
-			}
-			pmentry->dumpeable = dump_this_map (buff_smaps, pmentry->start_addr, pmentry->end_addr, pmentry->file_backed, pmentry->anonymous, pmentry->perms, filter_flags);
-			eprintf (" %"PFMT64x"-%"PFMT64x" - anonymous: %d, kernel_mapping: %d, file_backed: %d, dumpeable: %d\n\n",
-					pmentry->start_addr, pmentry->end_addr,
-					pmentry->anonymous, pmentry->kernel_mapping,
-					pmentry->file_backed, pmentry->dumpeable);
-
-			if (pmentry->file_backed) {
-				eprintf ("pmentry->name adding: %s as a SIZE_NT_FILE_DESCSZ\n", pmentry->name);
-				mapping_file.size += SIZE_NT_FILE_DESCSZ + len_name;
-				mapping_file.count++;
-			}
-		} else {
-			/* kernel mappings are always dumped */
-			pmentry->dumpeable = 1;
+		if (pmentry->name && !pmentry->kernel_mapping && !is_deleted) {
+			pmentry->file_backed = true;
+		}
+		pmentry->dumpeable = dump_this_map (buff_smaps, pmentry->start_addr, 
```

Comment:
wrong indent. you should only indent 2 tabs instead of 1 because its part of the previous definition